packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,474 @@
+  // (so an element with this id only exists when the tooltip is shown)
+  id={tooltipId}
+  calloutProps={calloutProps}
+  styles={hostStyles}
+>
+  <DefaultButton aria-describedby={tooltipId}>Hover over me</DefaultButton>
+</TooltipHost>
+```
+
+#### Drawbacks
+
+There are a few drawbacks with this approach to adding tooltips, which are outlined in [☂ Tooltip: open issues to resolve in converged approach #15102](https://github.com/microsoft/fluentui/issues/15102), and summarized below:
+
+- The wrapper `div` created by `TooltipHost` can cause layout issues for the component. It also doesn't always result in proper positioning for the tooltip.
+- The API is overly complex.
+- It's not possible to have a tooltip within a tooltip
+- There's no coordination between tooltips on a page. For example, moving the mouse between two elements with tooltips should cause the second tooltip to appear immediately without fading in/out.
+
+### Tooltips in v0/Northstar
+
+v0 tooltips use a `trigger` property to render the tooltip's target component. However, unlike v8 it does not create a wrapper `div` around the target component, but instead adds listeners to the target component's props.
+
+```jsx
+<Tooltip content="Example tooltip" trigger={<Button content="A button" />} />
+```
+
+# Sample Code
+
+There are a few ways to add a tooltip to a component.
+
+## A component with a `tooltip` slot
+
+Components can choose to expose a pseudo-slot `tooltip`, by inheriting props from the `WithTooltipSlot` interface and using the `useTooltipSlot` hook.
+
+While this looks and acts like a normal slot to users, the `Tooltip` is not actually rendered by the component, but rather passed to the `TooltipManager` to render. The `useTooltipSlot` hook is lightweight (small bundle size impact); it does not have a direct dependency on the `Tooltip` or `TooltipManager` classes.
+
+For example, to add the `tooltip` slot to Button:
+
+```typescript
+// Button.types.ts:
+export interface ButtonProps extends WithTooltipSlot /*...*/ {
+  // ...
+}
+
+// useButton.ts:
+export const useButton = (/*...*/) => {
+  const state = mergeProps(/*...*/);
+
+  // ...
+  useTooltipSlot(state);
+  // ...
+
+  return state;
+};
+```
+
+Example usage:
+
+```jsx
+<Button tooltip="Example tooltip" />
+<Button tooltip={<>Custom <b>Tooltip</b> content!</>} />
+<Button tooltip={{ children: 'Placed Tooltip', placement: 'right' }} />
+<Button tooltip={{ as: MyTooltip, children: 'You can even implement your own tooltip' }} />
+```
+
+## On any element using a ref
+
+To attach a tooltip to a component that doesn't have a `tooltip` slot, use the `useTooltipRef` hook. It takes the same shorthand props that the `tooltip` slot does. This is slightly less ergonomic than the tooltip slot, and requires using a ref, but works with any element.
+
+```jsx
+<div ref={useTooltipRef('Example tooltip', { id: 'exampleTooltipId' })} aria-describedby="exampleTooltipId">
```

Comment:
alternatively:

const tooltipProps = useTooltip('Example tooltip');
<div {...tooltipProps} />

Reply:
NIT: to make tooltips accessible, the trigger needs to be focusable, so it would probably be better to change the examples to include focusable elements only:
`<button ref={useTooltipRe} ... />

Reply:
one more thing to think through - how would the markup look like if the consumer needs to add tooltip, but also needs to get a ref to the element? Since here I think the useTooltipRef takes over the whole ref

Reply:
> how would the markup look like if the consumer needs to add tooltip, but also needs to get a ref to the element

Resolving another comment I made in favour of a single thread. I have a similar concern, will we oblige consumers to merge refs ?

Reply:
The problem with `useTooltip` returning props is that it could overwrite some existing props instead of merging them. For example, it'll replace any existing `onFocus` handler.

Ling and I talked today and I'm now pursuing an approach like a `<TooltipTrigger>` component that wraps another React component. It'll add/merge `onFocus`, `aria-describedby`, etc. to the child component's props, without creating a wrapper `div`. In that way it'd work similarly to the v0 Tooltip, except it'd delegate to TooltipManager to handle actually showing the tooltip.